cassandra-scheduler/src/main/java/com/mesosphere/dcos/cassandra/scheduler/resources/UpgradeSSTableRequest.java

```diff
@@ -0,0 +1,102 @@
+package com.mesosphere.dcos.cassandra.scheduler.resources;
+
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.mesosphere.dcos.cassandra.common.tasks.CassandraState;
+import com.mesosphere.dcos.cassandra.common.tasks.ClusterTaskRequest;
+import com.mesosphere.dcos.cassandra.common.tasks.upgradesstable.UpgradeSSTableContext;
+import com.mesosphere.dcos.cassandra.common.util.JsonUtils;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+public class UpgradeSSTableRequest implements ClusterTaskRequest {
+
+    public static final String ALL = "*";
+
+    @JsonCreator
+    public static UpgradeSSTableRequest create(
+            @JsonProperty("nodes") final List<String> nodes,
+            @JsonProperty("key_spaces") final List<String> keySpaces,
+            @JsonProperty("column_families") final List<String>
+                    columnFamilies) {
+        return new UpgradeSSTableRequest(nodes, keySpaces, columnFamilies);
+    }
+
+    @JsonProperty("nodes")
+    private final List<String> nodes;
+    @JsonProperty("key_spaces")
+    private final List<String> keySpaces;
+    @JsonProperty("column_families")
+    private final List<String> columnFamiles;
+
+    public UpgradeSSTableRequest(
+            final List<String> nodes,
+            final List<String> keySpaces,
+            final List<String> columnFamiles) {
+
+        this.nodes = (nodes == null) ? Collections.emptyList() : nodes;
+        this.keySpaces = (keySpaces == null) ? Collections.emptyList() :
+                keySpaces;
+        this.columnFamiles = (columnFamiles == null) ? Collections.emptyList() :
+                columnFamiles;
+    }
+
+
+    public List<String> getColumnFamiles() {
+        return columnFamiles;
+    }
+
+    public List<String> getKeySpaces() {
+        return keySpaces;
+    }
+
+    public List<String> getNodes() {
+        return nodes;
+    }
+
+    public boolean isValid(){
+        return !this.nodes.isEmpty();
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
```

Comment:
This is just general practice when you implement equals(), unless you know what you're doing here.